src/Test/test_nonlinear.jl

```diff
@@ -1115,3 +1115,560 @@ function setup_test(
     )
     return
 end
+
+function test_nonlinear_expression_hs071(
+    model::MOI.ModelLike,
+    config::MOI.Test.Config{T},
+) where {T}
+    @requires T == Float64
+    @requires _supports(config, MOI.optimize!)
+    F = MOI.ScalarNonlinearFunction
+    @requires MOI.supports(model, MOI.ObjectiveFunction{F}())
+    @requires MOI.supports_constraint(model, F, MOI.GreaterThan{Float64})
+    MOI.Utilities.loadfromstring!(
+        model,
+        """
+variables: w, x, y, z
+minobjective: ScalarNonlinearFunction(w * z * (w + x + y) + y)
```

Comment:
This is a big improvement for being able to test nlp solvers.

Reply:
Yeah. I was surprised how little modifications is required to the parsing code